### PR TITLE
TF 2.2 documentation update 

### PR DESCRIPTION
**Pull request by mchoi8739**
### Description of changes:
Update TF documentation, pointing out the newest feature with that latest Debugger pySDK and smdebug release.

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/tensorflow.md [hunk 1]
```diff
@@ -1,135 +1,238 @@
...
-hook = smd.KerasHook(out_dir=args.out_dir)
+hook = smd.KerasHook.create_from_json_file()
 
 model = tf.keras.models.Sequential([ ... ])
 model.compile(
     optimizer='adam',
     loss='sparse_categorical_crossentropy',
 )
 # Add the hook as a callback
+hook.set_mode(mode=smd.modes.TRAIN)
```
Comment (Vikas-kum) [inline docs/tensorflow.md:224]: comment here, hook.set_mode will help you differentiate between tensors in TRAIN and EVAL phases. DEtails about mode here[link]

**Comment by Vikas-kum** [comment]
On main doc page there is a Note: Limited support of the zero script change experience for TensorFlow 2.2. The tensor collections layers, inputs, outputs, and gradients are currently not available 

I think you would need to remove that.

**Comment by mchoi8739** [comment]
> On main doc page there is a Note: Limited support of the zero script change experience for TensorFlow 2.2. The tensor collections layers, inputs, outputs, and gradients are currently not available
> 
> I think you would need to remove that.

I will do after I verify by running the test notebook that I found the image issue with cuda version. I want to make sure if everything working is true for customers.

File: docs/api.md [hunk 2]
```diff
@@ -163,6 +163,7 @@ Note that `smd` import below translates to `import smdebug.{framework} as smd`.
...
 |`close()` | - | Closes all files that are currently open by the hook |
 | `save_scalar()` | `name (str)` <br/> `value (float)` <br/> `sm_metric (bool)`| Saves a scalar value by the given name. Passing `sm_metric=True` flag also makes this scalar available as a SageMaker Metric to show up in SageMaker Studio. Note that when `sm_metric` is False, this scalar always resides only in your AWS account, but setting it to True saves the scalar also on AWS servers. The default value of `sm_metric` for this method is False. |
+| `save_tensor()`| tensor_name (str), tensor_value (float), collections_to_write (str) | - | Manually save metrics tensors. The `record_tensor_value()` API is deprecated in favor or `save_tensor()`.|
```
[the remark this answers is not in the document]
Comment (NihalHarish) [reply docs/api.md:166]: `collections_to_write: str or list[str]`

[the remark this answers is not in the document]
Comment (NihalHarish) [reply docs/api.md:166]: When I render this in markdown, the behavior column has a single `-`.
Is this intentional?

**Comment by NihalHarish** [comment]
Can we also add a section on the version and behavior of smdebug shipped on different DLCs

File: docs/tensorflow.md [hunk 3]
```diff
@@ -1,135 +1,238 @@
...
-#### 1. Create a hook
 
- To create the hook constructor, add the following code.
+### Step 1: Create a hook <a name="create-a-hook"></a>
+
+ To create the hook constructor, add the following code to your training script. This enables the `smdebug` tools for TensorFlow and creates a TensorFlow `hook` object. When you run the `fit()` API for training, specify the smdebug `hook` as `callbacks`, as shown following:
 
 ```python
 import smdebug.tensorflow as smd
 hook = smd.{hook_class}.create_from_json_file()
```
[the remark this answers is not in the document]
Comment (mchoi8739) [reply docs/tensorflow.md:120]: That's what I broke down into the following subsections. Alternatively, I can remove that code snippet and keep the examples in the subsections.

File: docs/api.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mchoi8739) [reply docs/api.md:166]: > > On main doc page there is a Note: Limited support of the zero script change experience for TensorFlow 2.2. The tensor collections layers, inputs, outputs, and gradients are currently not available
> > I think you would need to remove that.
> 
> I will do after I verify by running the test notebook that I found the image issue with cuda version. I want to make sure if everything working is true for customers.

Confirm that this bug has fixed.

**Comment by mchoi8739** [comment]
Confirm that the bug has fixed. Removing the limitation note.

File: docs/api.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mchoi8739) [reply docs/api.md:166]: Fixed this.

**Comment by mchoi8739** [comment]
> Can we also add a section on the version and behavior of smdebug shipped on different DLCs

I am not sure what are the different behavior would be. I thought it works in the same way for all Debugger-supported DLCs. Can you tell me what are those behaviors?

**Comment by Vikas-kum** [comment]
> > On main doc page there is a Note: Limited support of the zero script change experience for TensorFlow 2.2. The tensor collections layers, inputs, outputs, and gradients are currently not available
> > I think you would need to remove that.
> 
> I will do after I verify by running the test notebook that I found the image issue with cuda version. I want to make sure if everything working is true for customers.

"want to make sure if everything working is true for customers." -> This should be done before the docs merged in and made public. 
When are you planning to do this? Any blockers?

**Comment by mchoi8739** [comment]
> > > On main doc page there is a Note: Limited support of the zero script change experience for TensorFlow 2.2. The tensor collections layers, inputs, outputs, and gradients are currently not available
> > > I think you would need to remove that.
> > 
> > 
> > I will do after I verify by running the test notebook that I found the image issue with cuda version. I want to make sure if everything working is true for customers.
> 
> "want to make sure if everything working is true for customers." -> This should be done before the docs merged in and made public. I hope it is working now.

yes, I confirmed that it works now. I tested with an example notebook.

**Comment by NihalHarish** [comment]
> > Can we also add a section on the version and behavior of smdebug shipped on different DLCs
> 
> I am not sure what are the different behavior would be. I thought it works in the same way for all Debugger-supported DLCs. Can you tell me what are those behaviors?

In the case of TF 2.x containers, inputs, outputs, gradients, and intermediate layer are only saved for the TF 2.2.0 container

**Comment by mchoi8739** [comment]
> > > Can we also add a section on the version and behavior of smdebug shipped on different DLCs
> > 
> > 
> > I am not sure what are the different behavior would be. I thought it works in the same way for all Debugger-supported DLCs. Can you tell me what are those behaviors?
> 
> In the case of TF 2.x containers, inputs, outputs, gradients, and intermediate layer are only saved for the TF 2.2.0 container

I made a note below the built-in collection list: https://github.com/mchoi8739/sagemaker-debugger/blob/master/docs/tensorflow.md#debugger-built-in-tensor-collections-for-tensorflow

File: docs/api.md [hunk 4]
```diff
@@ -163,6 +163,7 @@ Note that `smd` import below translates to `import smdebug.{framework} as smd`.
...
 |`close()` | - | Closes all files that are currently open by the hook |
 | `save_scalar()` | `name (str)` <br/> `value (float)` <br/> `sm_metric (bool)`| Saves a scalar value by the given name. Passing `sm_metric=True` flag also makes this scalar available as a SageMaker Metric to show up in SageMaker Studio. Note that when `sm_metric` is False, this scalar always resides only in your AWS account, but setting it to True saves the scalar also on AWS servers. The default value of `sm_metric` for this method is False. |
+| `save_tensor()`| tensor_name (str), tensor_value (float), collections_to_write (str or list[str]) | Manually save metrics tensors. The `record_tensor_value()` API is deprecated in favor or `save_tensor()`.|
```
[the remark this answers is not in the document]
Comment (mchoi8739) [reply docs/api.md]: fixed

File: README.md [hunk 5]
```diff
@@ -68,8 +68,6 @@ The following frameworks are available AWS Deep Learning Containers with the dee
 | [PyTorch](docs/pytorch.md) | 1.4, 1.5 |
```
Comment (NihalHarish) [inline README.md:68]: Please add Pytorch 1.6

Comment (NihalHarish) [reply README.md:68]: And TF 2.3.0

File: docs/api.md [hunk 6]
```diff
@@ -163,6 +163,7 @@ Note that `smd` import below translates to `import smdebug.{framework} as smd`.
...
 |`close()` | - | Closes all files that are currently open by the hook |
 | `save_scalar()` | `name (str)` <br/> `value (float)` <br/> `sm_metric (bool)`| Saves a scalar value by the given name. Passing `sm_metric=True` flag also makes this scalar available as a SageMaker Metric to show up in SageMaker Studio. Note that when `sm_metric` is False, this scalar always resides only in your AWS account, but setting it to True saves the scalar also on AWS servers. The default value of `sm_metric` for this method is False. |
+| `save_tensor()`| `tensor_name (str)`, `tensor_value (np.ndarray)`, `collections_to_write (str or list[str])` | Manually save metrics tensors. The `record_tensor_value()` API is deprecated in favor or `save_tensor()`.|
```
Comment (Vikas-kum) [inline docs/api.md:166]: numpy.array | numpy.ndarray

Comment (mchoi8739) [reply docs/api.md]: fixed
